scripts/scene/RoomBuilder.gd

```diff
@@ -15,6 +15,7 @@ var door_tileset_id: int = 4 # TEMPORARY. Door tiles will be included in room ti
 
 var initial_tile_coords = Vector2i()
 var transverse_tile_coords = Vector2i()
+var temp_door_coords: Array[Vector2i] = []
```

Comment:
I also considered initialising a `current_room` instead, and appending the door tile coords as we were doing before. Now we only create the new_room on confirming it. Otherwise, we would generate and append new rooms each time we went back to the SelectingTiles state.